pkg/prometheus/promcfg.go

```diff
@@ -2134,6 +2151,10 @@ func (cg *ConfigGenerator) generateK8SSDConfig(
 		})
 	}
 
+	if ptr.Deref(selectorMechanism, monitoringv1.SelectorMechanismRelabel) == monitoringv1.SelectorMechanismRole {
+		k8sSDConfig = generateRoleSelectorConfig(k8sSDConfig, selectorRoles, selector, cg)
```

Comment:
hmm I notice that we also inject the "selectors" key when running in daemonset mode. While DaemonSet mode doesn't support service monitors, it might in the future and it would get messy.